### PR TITLE
feat(challenge-parser): allow many questions in MC challenge

### PR DESCRIPTION
**Pull request by moT01**
Second attempt at this. 

It's a start to allowing many multiple choice questions on a single challenge. It adjusts the challenge parser to allow more questions by just adding additional `--text--`, `--answers--`, and `--video-solution--` headers - and puts all the questions in an array.

For now, the client just gets the first in the array and uses the rest of the existing code. I will make another PR to map over them and display them all.

To test: you can find any multiple choice challenge, add an additional questions by copy/pasting the question that's there - just the `text`, `answers` and `video-solution` part. To end up with something like this:

```
# --questions--

## --text--

When making a request from the Twitter API, what information must always be sent with the request?

## --answers--

Twitter username

---

date range

---

search term

---

key

## --video-solution--

4

## --text--

When making a request from the Twitter API, what information must always be sent with the request?

## --answers--

Twitter username

---

date range

---

search term

---

key

## --video-solution--

4
```

Then find the challenge in graphiql and check the questions.

One extra change I made was to move all the `--assignment--` headers that were nested in a question to outside of the question. Those were all in the Odin section.

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

Partially addresses https://github.com/freeCodeCamp/fCC10/issues/19

Todo:
-Make client handle many questions
-Update docs

### CODE REVIEW

**Comment by moT01** [comment]
Don't merge until we confer with mobile.

**Comment by ojeytonwilliams** [comment]
> Second attempt at this.

Does this replace https://github.com/freeCodeCamp/freeCodeCamp/pull/55884 or are you deciding between them?

**Comment by moT01** [comment]
It replaces it. The main difference being the way you create the questions in the markdown. The other one was to add another heading for each question, e.g: `--question1--`, `--question2`, etc, along with the subheadings. With this, you just need to add more sub-headings e.g: `--text--`, `-answers--`, `--video-solution--`

File: tools/challenge-parser/parser/plugins/add-video-question.js
Comment (ojeytonwilliams) [inline tools/challenge-parser/parser/plugins/add-video-question.js:31]:
```suggestion

      allQuestionNodes.forEach(questionNode => {
        const isStartOfQuestion =
          questionNode.children?.[0]?.value === '--text--';
        if (isStartOfQuestion) {
          questionTrees.push([questionNode]);
        } else {
          questionTrees[questionTrees.length - 1].push(questionNode);
        }
      });

```
I think this is a little cleaner and self-descriptive.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: tools/challenge-parser/parser/plugins/add-video-question.js
Comment (ojeytonwilliams) [reply tools/challenge-parser/parser/plugins/add-video-question.js:31]: I think this is the right place, but reviewing through VSCode is horrendous and reviewing through the browser is impossible :(

Anyways, it's supposed to replace the code that populates the `questionTrees`

File: curriculum/schema/challenge-schema.js
Comment (moT01) [reply curriculum/schema/challenge-schema.js]: ✅

File: tools/challenge-parser/parser/plugins/add-video-question.js
Comment (moT01) [reply tools/challenge-parser/parser/plugins/add-video-question.js]: ✅

**Comment by moT01** [comment]
This should be good to go. The mobile team added the ability to handle the new questions. I believe we will need to do a curriculum sync after this gets merged.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Comment by jdwilkin4** [comment]
@moT01 

Things look good.
Once the failing node tests for italian and portuguese are resolved then we should be good to merge this in

**Comment by moT01** [comment]
That's going to need a curriculum sync after this gets merged.

**Review by jdwilkin4 — APPROVED** [review]

**Pull request merged**
